### DISCUSSION TITLE
Unable to read remote logs with celery executor

### DISCUSSION THREAD

**Discussion by lohithasree** [apache/airflow#32063] [General]
### Apache Airflow version

Other Airflow 2 version (please specify below)

### What happened

We migrated from Local Executor to Celery Executor where using RabbitMQ as a broker.

With Local executor, remote logging is working as expected. Able to read logs

**Airflow Version: 2.6.1
Celery Version: 5.3.0**
```
apache-airflow-providers-common-sql==1.1.0
apache-airflow-providers-ftp==3.1.0
apache-airflow-providers-google==8.3.0
apache-airflow-providers-http==4.0.0
apache-airflow-providers-imap==3.0.0
apache-airflow-providers-postgres==5.2.0
apache-airflow-providers-celery==3.2.0
```
**Airflow Config for Remote Logging:**
```cfg
base_log_folder = /opt/airflow/logs
remote_logging = True
remote_log_conn_id = google_cloud_default
delete_local_logs = True
google_key_path = 
remote_base_log_folder = gs://bucket-name/logs
remote_task_handler_kwargs =
encrypt_s3_logs = False
logging_level = INFO
celery_logging_level =
fab_logging_level = WARNING
logging_config_class =
colored_console_log = False
colored_log_format = [%(blue)s%(asctime)s%(reset)s] {%(blue)s%(filename)s:%(reset)s%(lineno)d} %(log_color)s%(levelname)s%(reset)s - %(log_color)s%(message)s%(reset)s
colored_formatter_class = airflow.utils.log.colored_log.CustomTTYColoredFormatter
log_format = [%(asctime)s] {%(filename)s:%(lineno)d} %(levelname)s - %(message)s
simple_log_format = %(asctime)s %(levelname)s - %(message)s
dag_processor_log_target = file
dag_processor_log_format = [%(asctime)s] [SOURCE:DAG_PROCESSOR] {%(filename)s:%(lineno)d} %(levelname)s - %(message)s
log_formatter_class = airflow.utils.log.timezone_aware.TimezoneAware
secret_mask_adapter =
task_log_prefix_template =
log_filename_template = dag_id={{ ti.dag_id }}/run_id={{ ti.run_id }}/task_id={{ ti.task_id }}/{% if ti.map_index >= 0 %}map_index={{ ti.map_index }}/{% endif %}attempt={{ try_number }}.log
log_processor_filename_template = {{ filename }}.log
dag_processor_manager_log_location = /opt/airflow/logs/dag_processor_manager/dag_processor_manager.log
task_log_reader = task
extra_logger_names =
worker_log_server_port = 8793
trigger_log_server_port = 8794
file_task_handler_new_folder_permissions = 0o775
file_task_handler_new_file_permissions = 0o664
```

when we trigger dags, dags are going into queue state and then failed state by stating unable to read the remote logs like below error.

**Celery Config:**
```
celery_app_name = airflow.executors.celery_executor
worker_concurrency = 16
worker_prefetch_multiplier = 1
worker_enable_remote_control = True
broker_url = amqps://username:password@rabbitmq1-dev.project-name.dev.gcp.cloud.xx:5671/host_name;amqps://username:password@rabbitmq2-dev.project-name.dev.dev.gcp.cloud.xx:5671/host_name;amqps://username:password@rabbitmq3-dev.project-name.dev.gcp.cloud.xx:5671/host_name
flower_host = 0.0.0.0
flower_url_prefix = flower
flower_port = 5555
flower_basic_auth = QW3Q3uWPJqloOhEWv:pvmGvSATFxBMuD4El
sync_parallelism = 0
celery_config_options = airflow.config_templates.default_celery.DEFAULT_CELERY_CONFIG
ssl_active = True
ssl_key = /opt/airflow/cert/ddsdev-airflow.uk.hsbc.key
ssl_cert = /opt/airflow/cert/ddsdev-airflow.uk.hsbc.crt
ssl_cacert =
pool = prefork
operation_timeout = 1.0
task_track_started = True
task_publish_max_retries = 3
worker_precheck = False
result_backend = db+postgresql://username:password@localhost:5432/database
```
Accessing Airflow webserver and Flower UI using NGINX reverse proxy
```
***** Unable to read remote log from gs://**********/logs/dag_id=DDS_catalog_update/run_id=manual__2023-06-20T15:22:58.216561+00:00/task_id=insert_dds_ct_start_ts/attempt=1.log *** 404 GET https://storage.googleapis.com/download/storage/v1/b/*********/o/logs%2Fdag_id%3DDDS_catalog_update%2Frun_id%3Dmanual__2023-06-20T15%3A22%3A58.216561%2B00%3A00%2Ftask_id%3Dinsert_dds_ct_start_ts%2Fattempt%3D1.log?alt=media: No such object: *********88/logs/dag_id=DDS_catalog_update/run_id=manual__2023-06-20T15:22:58.216561+00:00/task_id=insert_dds_ct_start_ts/attempt=1.log: ('Request failed with status code', 404, 'Expected one of', <HTTPStatus.OK: 200>, <HTTPStatus.PARTIAL_CONTENT: 206>) *** Could not read served logs: Request URL is missing an 'http://' or 'https://' protocol.**
```
We have verified the object path manually, Its existing. But from airflow we are getting error like above.

We have verified all the permissions and config.

Can you please help us to resolve.

### What you think should happen instead

I expected I could read the task log on UI

### How to reproduce

While I run any DAGs, the log always shows up this.
```
***** Unable to read remote log from gs://**********/logs/dag_id=DDS_catalog_update/run_id=manual__2023-06-20T15:22:58.216561+00:00/task_id=insert_dds_ct_start_ts/attempt=1.log *** 404 GET https://storage.googleapis.com/download/storage/v1/b/*********/o/logs%2Fdag_id%3DDDS_catalog_update%2Frun_id%3Dmanual__2023-06-20T15%3A22%3A58.216561%2B00%3A00%2Ftask_id%3Dinsert_dds_ct_start_ts%2Fattempt%3D1.log?alt=media: No such object: *********88/logs/dag_id=DDS_catalog_update/run_id=manual__2023-06-20T15:22:58.216561+00:00/task_id=insert_dds_ct_start_ts/attempt=1.log: ('Request failed with status code', 404, 'Expected one of', <HTTPStatus.OK: 200>, <HTTPStatus.PARTIAL_CONTENT: 206>) *** Could not read served logs: Request URL is missing an 'http://' or 'https://' protocol.**
```
The log files have already exists on worker and store on GCS.

### Operating System

RHEL 7

### Deployment

Virtualenv installation

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

**Comment by boring-cyborg** [comment]
Thanks for opening your first issue here! Be sure to follow the issue template! If you are willing to raise PR to address this issue please do so, no need to wait for approval.

**Comment by hussein-awala** [comment]
This might not be related to remote logging since the task fails immediately after being sent to the worker. I believe there could be an issue with the parsing of DAG files or the availability of these files in the workers' pods/containers/processes. Have you checked the workers' logs?

I convert the issue to discussion, and we can reopen it if needed.

**Reply by lohithasree** [reply]
Yes, I have verified the worker logs sometimes Its working and sometimes its not working. Not able to find the root cause.

**Comment by lohithasree** [comment]
I have attached the worker logs over here.

started celery worker in Airflow VM as Systemd.


[Unit]
Description=Airflow Celery Flower UI
After=network.target cloud-sql-proxy.service airflow-scheduler.service

[Service]
Environment="AIRFLOW_HOME=/opt/airflow"
EnvironmentFile=/etc/sysconfig/airflow
User=root
Group=root
Type=simple
ExecStart=/usr/local/bin/airflow celery worker
Restart=always
RestartSec=5s

[Install]
WantedBy=multi-user.target



● celery-worker.service - Airflow Celery Flower UI
   Loaded: loaded (/usr/lib/systemd/system/celery-worker.service; enabled; vendor preset: disabled)
   Active: active (running) since Wed 2023-06-21 08:47:50 UTC; 20h ago
 Main PID: 3040 ([celeryd: celer)
   CGroup: /system.slice/celery-worker.service
           ├─3040 [celeryd: celery@airflow-dds-dev2:MainProcess] -active- (celery worker
           ├─3104 gunicorn: master [gunicorn]
           ├─3107 gunicorn: worker [gunicorn]
           ├─3108 gunicorn: worker [gunicorn]
           ├─3149 [celeryd: celery@airflow-dds-dev2:ForkPoolWorker-1]
           ├─3154 [celeryd: celery@airflow-dds-dev2:ForkPoolWorker-2]
           ├─3156 [celeryd: celery@airflow-dds-dev2:ForkPoolWorker-3]
           ├─3161 [celeryd: celery@airflow-dds-dev2:ForkPoolWorker-4]
           ├─3164 [celeryd: celery@airflow-dds-dev2:ForkPoolWorker-5]
           ├─3166 [celeryd: celery@airflow-dds-dev2:ForkPoolWorker-6]
           ├─3168 [celeryd: celery@airflow-dds-dev2:ForkPoolWorker-7]
           ├─3172 [celeryd: celery@airflow-dds-dev2:ForkPoolWorker-8]
           ├─3188 [celeryd: celery@airflow-dds-dev2:ForkPoolWorker-9]
           ├─3194 [celeryd: celery@airflow-dds-dev2:ForkPoolWorker-10]
           ├─3200 [celeryd: celery@airflow-dds-dev2:ForkPoolWorker-11]
           ├─3206 [celeryd: celery@airflow-dds-dev2:ForkPoolWorker-12]
           ├─3212 [celeryd: celery@airflow-dds-dev2:ForkPoolWorker-13]
           ├─3214 [celeryd: celery@airflow-dds-dev2:ForkPoolWorker-14]
           ├─3215 [celeryd: celery@airflow-dds-dev2:ForkPoolWorker-15]
           └─3221 [celeryd: celery@airflow-dds-dev2:ForkPoolWorker-16]

Jun 22 05:19:12 airflow-dds-dev2 airflow[3040]: [2023-06-22 05:19:12,911: INFO/ForkPoolWorker-16] Task airflow.executors.celery_executor.execute_command[8fc7be32-3a66-4e23-bc87-ca9945c4d541] succeeded in 132.22622969000076s: None
Jun 22 05:19:49 airflow-dds-dev2 airflow[3040]: [2023-06-22 05:19:49,098: INFO/MainProcess] Task airflow.executors.celery_executor.execute_command[26fad01c-2563-442d-88f6-80510c9c9ee3] received
Jun 22 05:19:49 airflow-dds-dev2 airflow[3040]: [2023-06-22 05:19:49,114: INFO/ForkPoolWorker-16] [26fad01c-2563-442d-88f6-80510c9c9ee3] Executing command in Celery: ['airflow', 'tasks', 'run', 'DDS_ID_UK_PARTY_cust_acct_id_num_lkup', 'final_status', 'manual__2023-06-22T05:16:59.938208+00:00', '--local', '--subdir', 'DAGS_FOLDER/DDS_ID_UK_PARTY_cust_acct_id_num_lkup.py']
Jun 22 05:19:49 airflow-dds-dev2 airflow[3040]: [2023-06-22 05:19:49,271: INFO/ForkPoolWorker-16] Filling up the DagBag from /opt/airflow/dds_airflow/dags/DDS_ID_UK_PARTY_cust_acct_id_num_lkup.py
Jun 22 05:19:50 airflow-dds-dev2 airflow[3040]: [2023-06-22 05:19:50,086: WARNING/ForkPoolWorker-16] /usr/local/lib/python3.9/site-packages/airflow/models/baseoperator.py:429: RemovedInAirflow3Warning: provide_context is deprecated as of 2.0 and is no longer required
Jun 22 05:19:50 airflow-dds-dev2 airflow[3040]: result = func(self, **kwargs, default_args=default_args)
Jun 22 05:19:50 airflow-dds-dev2 airflow[3040]: [2023-06-22 05:19:50,420: WARNING/ForkPoolWorker-16] Changing /opt/airflow/logs/dag_id=DDS_ID_UK_PARTY_cust_acct_id_num_lkup/run_id=manual__2023-06-22T05:16:59.938208+00:00/task_id=final_status permission to 509
Jun 22 05:19:50 airflow-dds-dev2 airflow[3040]: [2023-06-22 05:19:50,542: INFO/ForkPoolWorker-16] Running <TaskInstance: DDS_ID_UK_PARTY_cust_acct_id_num_lkup.final_status manual__2023-06-22T05:16:59.938208+00:00 [queued]> on host airflow-dds-dev2.c.hsbc-11374226-ddsuk-dev.internal
Jun 22 05:19:53 airflow-dds-dev2 airflow[3040]: [2023-06-22 05:19:53,062: INFO/ForkPoolWorker-16] Previous log discarded: 404 GET https://storage.googleapis.com/download/storage/v1/b/ddsuk-dev-airflow-log/o/logs%2Fdag_id%3DDDS_ID_UK_PARTY_cust_acct_id_num_lkup%2Frun_id%3Dmanual__2023-06-22T05%3A16%3A59.938208%2B00%3A00%2Ftask_id%3Dfinal_status%2Fattempt%3D1.log?alt=media: No such object: ddsuk-dev-airflow-log/logs/dag_id=DDS_ID_UK_PARTY_cust_acct_id_num_lkup/run_id=manual__2023-06-22T05:16:59.938208+00:00/task_id=final_status/attempt=1.log: ('Request failed with status code', 404, 'Expected one of', <HTTPStatus.OK: 200>, <HTTPStatus.PARTIAL_CONTENT: 206>)
Jun 22 05:19:53 airflow-dds-dev2 airflow[3040]: [2023-06-22 05:19:53,229: INFO/ForkPoolWorker-16] Task airflow.executors.celery_executor.execute_command[26fad01c-2563-442d-88f6-80510c9c9ee3] succeeded in 4.127845276991138s: None
